kythe/extractors/openjdk11/extract/extract.go

```diff
@@ -0,0 +1,196 @@
+	kytheRootVar        = "KYTHE_ROOT_DIRECTORY"
+	kytheVnameVar       = "KYTHE_VNAMES"
+	javaMakeVar         = "JAVA_CMD"
+	runfilesWrapperPath = "io_kythe/kythe/extractors/openjdk11/java_wrapper"
+	runfilesVnamesPath  = "io_kythe/kythe/extractors/openjdk11/vnames.json"
+)
+
+var (
+	wrapperPath  string
+	vnameRules   string
+	errorPattern = regexp.MustCompile("ERROR: extractor failure for module ([^:]*):")
+)
+
+func setupRunfiles() error {
+	if os.Getenv("RUNFILES_DIR") != "" || os.Getenv("RUNFILES_MANIFEST_FILE") != "" {
+		return nil
+	}
+	for _, base := range []string{os.Args[0] + ".runfiles", "."} {
+		root, err := filepath.Abs(base)
+		if err != nil {
+			continue
+		} else if _, err := os.Stat(root); err != nil {
+			continue
+		}
+		os.Setenv("RUNFILES_DIR", root)
+		manifest := filepath.Join(root, "MANIFEST")
+		if _, err := os.Stat(manifest); err == nil {
+			os.Setenv("RUNFILES_MANIFEST_FILE", manifest)
+		}
+		return nil
+	}
+	return errors.New("unable to setup runfiles")
+}
+
+func defaultWrapperPath() string {
+	val, _ := bazel.Runfile(runfilesWrapperPath)
+	return val
+}
+
+func defaultVnamesPath() string {
+	val, _ := bazel.Runfile(runfilesVnamesPath)
+	return val
+}
+
+func makeDir() string {
+	if dir := flag.Arg(0); dir != "" {
+		return dir
+	}
+	return "."
+}
+
+func findJavaCommand() (string, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	cmd := exec.CommandContext(ctx, "make", "-n", "-p")
+	cmd.Dir = makeDir()
+	cmd.Stderr = os.Stderr
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return "", err
+	}
+	if err := cmd.Start(); err != nil {
+		return "", err
+	}
+	prefix := javaMakeVar + " := "
+	scanner := bufio.NewScanner(stdout)
+	for scanner.Scan() {
+		if strings.HasPrefix(scanner.Text(), prefix) {
+			cancel() // Safe to call repeatedly.
+			cmd.Wait()
+			return strings.TrimPrefix(scanner.Text(), prefix), nil
+		}
+	}
+	return "", cmd.Wait()
+}
+
+func mustFindJavaCommand() string {
+	java, err := findJavaCommand()
+	if err != nil {
+		log.Fatalf("unable to determine KYTHE_JAVA_COMMAND: %v", err)
+	}
+	return java
+}
+
+func setEnvDefaultFunc(env []string, key string, value func() string) []string {
+	if val := os.Getenv(key); len(val) == 0 {
+		env = append(env, key+"="+value())
+	}
+	return env
+}
+
+func makeEnv() []string {
+	env := os.Environ()
+	env = setEnvDefaultFunc(env, javaCommandVar, mustFindJavaCommand)
+	env = setEnvDefaultFunc(env, kytheRootVar, makeDir)
+	if len(vnameRules) > 0 {
+		env = setEnvDefaultFunc(env, kytheVnameVar, func() string { return vnameRules })
+	}
+	return env
+}
+
+func init() {
+	setupRunfiles()
+	flag.StringVar(&wrapperPath, "java-wrapper", defaultWrapperPath(), "path to the java_wrapper executable (optional)")
+	flag.StringVar(&vnameRules, "rules", defaultVnamesPath(), "path of vnames.json file (optional)")
+	flag.Usage = flagutil.SimpleUsage("Extract a configured openjdk11 source directory", "[--java-wrapper=] [path]")
+}
```

Comment:
Add a flag for the output directory and set the `KYTHE_OUTPUT_DIRECTORY` for `java_wrapper`.

I think adding as many of the `java_wrapper` env var parameters as possible as flags to this binary will make things cleaner.